File: 13.ListView/MyListView/app/src/main/java/com/dicoding/picodiploma/mylistview/MainActivity.java
package com.dicoding.picodiploma.mylistview;

import android.content.res.TypedArray;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    String[] dataName;
    String[] dataDescription;
    TypedArray imagePahlawan;
    PahlawanAdapter adapter;
    ListView listView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        listView = findViewById(R.id.lv_list);

        prepare();

        addItem();

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                Toast.makeText(MainActivity.this, dataName[i], Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void prepare() {
        dataName = getResources().getStringArray(R.array.data_name);
        dataDescription = getResources().getStringArray(R.array.data_description);
        imagePahlawan = getResources().obtainTypedArray(R.array.data_photo);
    }

    private void addItem() {
        ArrayList<Pahlawan> pahlawan = new ArrayList<>();
        adapter = new PahlawanAdapter(this, pahlawan);

        for (int i = 0; i < dataName.length; i++) {
            Pahlawan pahlawans = new Pahlawan();
            pahlawans.setImage(imagePahlawan.getResourceId(i, -1));
            pahlawans.setName(dataName[i]);
            pahlawans.setDescription(dataDescription[i]);
            pahlawan.add(pahlawans);
            adapter.setPahlawans(pahlawan);
        }
        listView.setAdapter(adapter);
    }
}

File: ListView/MyListView/kotlin/app/src/main/java/com/dicoding/picodiploma/mylistview/MainActivity.kt
package com.dicoding.picodiploma.mylistview

import android.content.res.TypedArray
import android.os.Bundle
import android.widget.AdapterView
import android.widget.ListView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var adapter: HeroAdapter
    private lateinit var dataName: Array<String>
    private lateinit var dataDescription: Array<String>
    private lateinit var dataPhoto: TypedArray
    private var heroes = arrayListOf<Hero>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val listView: ListView = findViewById(R.id.lv_list)

        adapter = HeroAdapter(this)

        listView.adapter = adapter

        //Menyipakan data dari resource
        prepare()

        //Menambahkan data dari resource ke adapter
        addItem()

        //Memberi aksi pada lv_list
        listView.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
            Toast.makeText(this@MainActivity, heroes[position].name, Toast.LENGTH_SHORT).show()
        }
        
    }

    private fun prepare() {
        dataName = resources.getStringArray(R.array.data_name)
        dataDescription = resources.getStringArray(R.array.data_description)
        dataPhoto = resources.obtainTypedArray(R.array.data_photo)
    }

    private fun addItem() {
        for (position in dataName.indices) {
            val hero = Hero(
                    dataPhoto.getResourceId(position, -1),
                    dataName[position],
                    dataDescription[position]
            )
            heroes.add(hero)
        }
        adapter.heroes = heroes
    }
}
